src/retryOnEmpty.test.ts

```diff
@@ -725,14 +708,14 @@ function stubGenericRequest<T extends JsonRpcParams, U extends Json>(
 }
 
 /**
- * Builds a canned response for a request made to `provider.sendAsync` which
+ * Builds a canned response for a request made to `provider.request` which
  * will error for the first N instances and then succeed on the last instance.
  * Intended to be used in conjunction with `stubProviderRequests`.
  *
  * @param request - The request matcher for the stub.
  * @param numberOfTimesToFail - The number of times the request is expected to
  * be called until it returns a successful response.
- * @param successfulResponse - The response that `provider.sendAsync` will
+ * @param successfulResponse - The response that `provider.request` will
```

Comment:
Since this is no longer expected to return a response object, should we call this `successfulResult` instead?